opba-banking-protocol-facade/src/main/java/de/adorsys/opba/protocol/facade/services/ServiceContextProvider.java

```diff
@@ -58,38 +79,79 @@
                 .build();
     }
 
+    @NotNull
+    @SneakyThrows
+    private ContextWithKey getEncryptedContext(FacadeServiceableRequest facadeServiceable,
+                                               EncryptionService encryptionService,
+                                               KeyDTO keyDTO) {
+        String encryptedContext = new String(encryptionService.encrypt(MAPPER.writeValueAsBytes(facadeServiceable)));
+        return new ContextWithKey(encryptedContext, keyDTO);
+    }
+
     private <T extends FacadeServiceableGetter> ServiceSession extractOrCreateServiceSession(
             T request,
-            AuthSession authSession
+            AuthSession authSession,
+            Supplier<ContextWithKey> encryptedContext
     ) {
         if (null != authSession) {
             return authSession.getParent();
         } else {
-            return findServiceSessionByIdOrCreate(request);
+            return findServiceSessionByIdOrCreate(request, encryptedContext);
         }
     }
 
     @NotNull
     @SneakyThrows
-    private <T extends FacadeServiceableGetter> ServiceSession findServiceSessionByIdOrCreate(T request) {
-        if (null != request.getFacadeServiceable().getServiceSessionId()) {
-            Optional<ServiceSession> existingSession = serviceSessions
-                    .findById(request.getFacadeServiceable().getServiceSessionId());
-            if (existingSession.isPresent()) {
-                return existingSession.get();
+    private <T extends FacadeServiceableGetter> ServiceSession findServiceSessionByIdOrCreate(
```

Comment:
This method always creates new ServiceSession, but it has name `findServiceSessionByIdOrCreate` is it correct?

Reply:
I split method but the name remains

Reply:
renamed to createServiceSession